ledger/src/blockstore.rs

```diff
@@ -7181,26 +7181,30 @@ pub mod tests {
         ));
         let term_last_idx = terminator.last().unwrap().index() as usize;
         // Trying to insert another "is_last" shred with index < the received index should fail
-        // skip over shred 7
+        // so we insert some shreds that are beyond where terminator block is in index space
         blockstore
             .insert_shreds(
-                shreds[term_last_idx + 2..term_last_idx + 4].iter().cloned(),
+                shreds[term_last_idx + 2..term_last_idx + 3].iter().cloned(),
                 None,
                 false,
             )
             .unwrap();
         let slot_meta = blockstore.meta(0).unwrap().unwrap();
-        assert_eq!(slot_meta.received, term_last_idx as u64 + 4);
+        assert_eq!(slot_meta.received, term_last_idx as u64 + 3);
         let mut duplicate_shreds = vec![];
-        assert!(!blockstore.should_insert_data_shred(
-            terminator.last().unwrap(),
-            &slot_meta,
-            &HashMap::new(),
-            max_root,
-            None,
-            ShredSource::Repaired,
-            &mut duplicate_shreds,
-        ));
+        // Now we check if terminator is eligible to be inserted
+        assert!(
+            !blockstore.should_insert_data_shred(
+                terminator.last().unwrap(),
+                &slot_meta,
+                &HashMap::new(),
+                max_root,
+                None,
+                ShredSource::Repaired,
+                &mut duplicate_shreds,
+            ),
+            "Should not insert shreds with indices before already existing shreds"
```

Comment:
The key part is this shred has the "last" flag set. We insert shreds w/ out of order index all the time.
`"Should not insert shred with last flag set and index less than already existing shreds"`

Reply:
Fixed in https://github.com/anza-xyz/agave/pull/5731/commits/6df02fd66bac92fa52c09489ebbdfed500325341